src/main/java/seedu/duke/command/ArchiveNoteCommand.java

```diff
@@ -47,31 +51,27 @@ public ArchiveNoteCommand(String title) {
 
     @Override
     public String execute() {
-        boolean isArchived;
-        String archivedTitle = "";
+        String archivedTitle;
         Note archivedNote;
-        ArrayList<Note> archivedNotesList = new ArrayList<>();
+        ArrayList<Note> archivedNotesList;
 
         try {
             // If there is no title, archive note by index. Else archive by title.
             if (title.isBlank()) {
                 archivedNote = notebook.getNotes().get(index);
                 archivedTitle = archivedNote.getTitle();
 
-                archivedNotebook.addNote(archivedNote);
-                isArchived = notebook.deleteNote(index);
+                archivedNote.toggleIsArchived();
 
                 return Formatter.formatString(ARCHIVE_NOTE_MESSAGE + archivedTitle);
 
-            } else if (notebook.getNote(title)) {
+            } else {
                 archivedNotesList = (ArrayList<Note>) notebook.getNotes().stream()
                         .filter((s) -> s.getTitle().equalsIgnoreCase(title))
                         .collect(Collectors.toList());
```

Comment:
Why is there a need for archivedNotesList? Since all the note can only have unique names, wont it only return 1 note object? There is a getNote(title) function in notebook, you can use that instead.

Reply:
getNotetitle() returned a boolean, I have changed it but I have decided to overload it.